### PR TITLE
Comprehensive syntax code check

### PR DESCRIPTION
**Pull request by rienafairefr**
When starting tox, this will 
* start nosetests and fail the build if this fails
* examine the source code with pylint and flake8 and fail on any syntax problem

Failing a build because of a syntax problem might be pedantic, discussions welcome

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -27,15 +27,14 @@ deps = -rtests_utils/test-requirements.txt
 commands =
     python setup.py nosetests {posargs}
```
Comment (aabadie) [inline tox.ini]: This toxenv looks useless to me since code_check and test envs already do the same things.

File: tox.ini [hunk 2]
```diff
@@ -27,15 +27,14 @@ deps = -rtests_utils/test-requirements.txt
...
     bash -c "python setup.py lint | tee pylint.out || true"
-    bash -c "python setup.py pep8 | tee pep8.out || true"
-
+    bash -c "flake8 | tee flake8.out || true"
```
Comment (aabadie) [inline tox.ini]: I would keep the pep8 and add the flake8 check

File: roomba/roomba.py [hunk 3]
```diff
@@ -769,7 +769,7 @@ def interface_serial(self):
             try:
                 if self.qsens.qsize > 0:
                     self.qsens.get_nowait()
-            except:
+            except:  # noqa: E722
```
Comment (aabadie) [inline roomba/roomba.py]: I think this can be excluded globally from the setup.cfg file. What is the humanly readable error type associated to it?

**Review by aabadie — CHANGES_REQUESTED** [review]
Good initiative ! Just a few comments below.

File: tox.ini (hunk 1 above)
Comment (rienafairefr) [reply tox.ini]: I agree, it was gaetan's way. I'll remove it if we are two to think it's useless :-)

File: roomba/roomba.py (hunk 3 above)
Comment (rienafairefr) [reply roomba/roomba.py]: OK. I'll add 

    # E722 do not use bare except
    ignore = E722

 in flake8 config

File: tox.ini (hunk 2 above)
Comment (rienafairefr) [reply tox.ini]: done, it should pass in travis

**Review by aabadie — APPROVED** [review]

**Comment by aabadie** [comment]
All good here. Let's merge this. Thanks @rienafairefr !

**Comment by rienafairefr** [comment]
"Your code has been rated at 10.00/10", et pas le droit de baisser en qualité :-)

**Comment by aabadie** [comment]
> "Your code has been rated at 10.00/10"

with the help of `#pylint:disable` tricks ;)

**Pull request merged**